File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP=0
        l=0
        r=1
        n=len(prices)
        while r<n:
            if prices[l]<prices[r]:
                maxP=max(maxP,prices[r]-prices[l])
            else:
                l=r
            r+=1
        return maxP

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length < 2) return nums.length;
        Arrays.sort(nums);
        int consec=1;
        int lastNum=nums[0];
        int maxConsec=1;
        for(int i = 1; i < nums.length; i++) {
            int curNum = nums[i];
            if(lastNum==curNum) continue;
            if(lastNum+1==curNum) {
                consec++;
            } else {
                consec=1;
            }
            lastNum=curNum;
            maxConsec = Math.max(maxConsec,consec);
        }
        return maxConsec;
    }
}
/*

*/